### ISSUE TITLE
Accessibility in Header Right Panel

### ISSUE THREAD

**Issue by barbarabeat** [carbon-design-system/carbon-components-vue#1346]
## Detailed description

> Describe in detail the issue you're having.

When using cv-header with right panel, if there is some element that can be expanded, such as a switcher, if run 'IBM Equal Access Accessibility Checker ' there is some accessibility errors about, including violations. Also, if I use the keyboard and my right panel is not expanded, I still can "walk" there.

> Is this a feature request (new component, new icon), a bug, or a general issue?

Accessibility issue

> Is this issue related to a specific component?

cv-header right panel

> What did you expect to happen? What happened instead? What would you like to see changed?

 If run 'IBM Equal Access Accessibility Checker ', it should not appears accessibility errors for right panel component. Also, If I use the keyboard, and my panel is not expanded, it should go to the next component, not inside my panel.

> What browser are you working in?

Firefox and chrome

> What version of the Carbon Design System are you using?

Latest

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

--

## Steps to reproduce the issue

1. Go to some header with right panel, such as (https://vue.carbondesignsystem.com/?path=/story/components-cvuishell--header-base-with-actions-and-right-panels)
2. Run 'IBM Equal Access Accessibility Checker '
3. You will find some accessibility errors because of this componenet. Such as "Element "a" should not be focusable within the subtree of an element with an 'aria-hidden' attribute with value 'true'"
4. Select the header and use the keyboard to walk into the page. If using the link above, in the last component in the header, if you do not expand it, you still will "walk" there.

## Additional information

- Screenshots or code
- Notes
<img width="1434" alt="Captura de Tela 2022-07-19 às 13 09 15" src="https://user-images.githubusercontent.com/25625728/179806830-9ebd32a6-4e62-4d7f-acaf-55af5682111d.png">

**Comment by davidnixon** [comment]
This was originally written against Vue 2 components but is a good candidate for an accessibility test for the `src/components/CvUIShell/CvHeaderPanel.vue`  component in Vue 3. See [guidance](https://github.com/carbon-design-system/carbon-components-vue/blob/main/ACCESSIBILITY-TESTING.md)
